File: src/Scripting/PHP/patterns/active_record.php
<?php declare(strict_types=1);
$check = static fn(bool $ok) => $ok ?: throw new RuntimeException('active_record');
$record = new class(1,'Ada') {
    private static array $db=[];
    public function __construct(public int $id, public string $name) {}
    public function save(): void { self::$db[$this->id] = $this->name; }
    public static function findName(int $id): ?string { return self::$db[$id] ?? null; }
};
$record->save(); $check($record::findName(1) === 'Ada');

File: src/Scripting/Ruby/patterns/active_record.rb
# frozen_string_literal: true

module RubyPatterns
  module ActiveRecord
    module_function

    def run
      table = {}
      user_class = Class.new do
        define_method(:initialize) { |id, name, storage| @id, @name, @storage = id, name, storage }
        define_method(:save) { @storage[@id] = { name: @name } }
      end
      user_class.new(1, 'Ada', table).save
      raise 'Active Record failed' unless table[1][:name] == 'Ada'
      true
    end
  end
end

RubyPatterns::ActiveRecord.run if $PROGRAM_NAME == __FILE__
